### ISSUE TITLE
Use OFFSET macro in asm-offsets.c

### ISSUE THREAD

**Issue by mpe** [linuxppc/linux#44]
All of our entries in asm-offests.c look like this:

```
DEFINE(TI_FLAGS, offsetof(struct thread_info, flags));
```

But there is a common macro, `OFFSET`, which makes this cleaner, eg:

```
OFFSET(TI_flags, thread_info, flags);
```

**Comment by csmart** [comment]
How does one address lines like this?
DEFINE(GPR0, STACK_FRAME_OVERHEAD+offsetof(struct pt_regs, gpr[0]));

**Comment by mpe** [comment]
Good question. If it's just one or two that use addition we can just continue to use `DEFINE`.

Given how many we have that use `STACK_FRAME_OVERHEAD` it's probably worth locally declaring a macro that does that, eg. something like:

```
#define STACK_OFFSET(_x) DEFINE(_x, STACK_FRAME_OVERHEAD + offsetof(struct pt_regs, _x))
```

I'd also like to eventually rename all the ones that use `STACK_FRAME_OVERHEAD`, ie. are accessing the `pt_regs` on the stack, to make it clear they're doing that. But that should be a separate issue.

**Comment by mpe** [comment]
Implemented in 454656155110. Thanks @RashmicaG  :champagne:
